.github/actions/python-setup/action.yaml

```diff
@@ -0,0 +1,27 @@
+name: "Python setup"
+description: |
+  Runs an opinionated and unified python setup action. It does the following:
+  * Installs python
+  * Installs poetry
+  * Installs a specified poetry project, if given
+inputs:
+  POETRY_DIRECTORY:
```

Comment:
might be clearer to call this `pyproject_location` location or something since this doesn't actually refer to the location of poetry.